pkg/controller/perconaservermongodb/mgo.go

```diff
@@ -99,11 +100,11 @@ func (r *ReconcilePerconaServerMongoDB) reconcileCluster(cr *api.PerconaServerMo
 		cnf.Version++
 		err = mongo.WriteConfig(session, cnf)
 		if err != nil {
-			return errors.Wrap(err, "add new: write mongo config")
+			return false, errors.Wrap(err, "add new: write mongo config")
 		}
 	}
 
-	return nil
+	return len(session.LiveServers()) == len(pods.Items), nil
```

Comment:
we are migrating to another mongo driver in https://github.com/percona/percona-server-mongodb-operator/pull/368/

it will work with new driver?

Reply:
No, it won't.
@AlyHKafoury we should extract "live nodes" from isMaster https://docs.mongodb.com/manual/reference/command/isMaster/#dbcmd.isMaster

Reply:
@delgod @dAdAbird I will wait then for the other branch merge and rebase off it to update the rest of the branch with the new driver